### ISSUE TITLE
Problem while deleting the dumped data from MongoDB

### ISSUE THREAD

**Issue by hugofpaiva** [doableware/djongo#538]
Although everything works fine on my local machine, while doing the Django Tests with GitHub Actions there is an error while destroying the database created for the tests. Right on the last line of the Djongo code when it is deleting the db on `creation.py` at line 30, Python can't delete the directory tree where the dumped data supposedly is with the path `dump`.

I am using Python version `3.9` and the MongoDB Docker image `mongo:4.4.4`.

#### Traceback
<img width="1259" alt="Screenshot 2021-04-14 at 17 25 32" src="https://user-images.githubusercontent.com/44342122/114745294-6d2efe80-9d46-11eb-9a4e-6714dad2352f.png">

```
Run make test
env/bin/python manage.py test --verbosity=0 --parallel --failfast
This version of djongo does not support "NULL, NOT NULL column validation check" fully. Visit https://nesdis.github.io/djongo/support/
Traceback (most recent call last):
This version of djongo does not support "schema validation using CONSTRAINT" fully. Visit https://nesdis.github.io/djongo/support/
  File "/home/runner/work/rest_api/rest_api/manage.py", line 22, in <module>
This version of djongo does not support "schema validation using KEY" fully. Visit https://nesdis.github.io/djongo/support/
    main()
This version of djongo does not support "schema validation using REFERENCES" fully. Visit https://nesdis.github.io/djongo/support/
  File "/home/runner/work/rest_api/rest_api/manage.py", line 18, in main
This version of djongo does not support "COLUMN DROP NOT NULL " fully. Visit https://nesdis.github.io/djongo/support/
This version of djongo does not support "DROP CASCADE" fully. Visit https://nesdis.github.io/djongo/support/
    execute_from_command_line(sys.argv)
  File "/home/runner/work/rest_api/rest_api/env/lib/python3.9/site-packages/django/core/management/__init__.py", line 401, in execute_from_command_line
    utility.execute()
  File "/home/runner/work/rest_api/rest_api/env/lib/python3.9/site-packages/django/core/management/__init__.py", line 395, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/home/runner/work/rest_api/rest_api/env/lib/python3.9/site-packages/django/core/management/commands/test.py", line 23, in run_from_argv
    super().run_from_argv(argv)
  File "/home/runner/work/rest_api/rest_api/env/lib/python3.9/site-packages/django/core/management/base.py", line 328, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/home/runner/work/rest_api/rest_api/env/lib/python3.9/site-packages/django/core/management/base.py", line 369, in execute
    output = self.handle(*args, **options)
  File "/home/runner/work/rest_api/rest_api/env/lib/python3.9/site-packages/django/core/management/commands/test.py", line 53, in handle
    failures = test_runner.run_tests(test_labels)
  File "/home/runner/work/rest_api/rest_api/env/lib/python3.9/site-packages/django/test/runner.py", line 684, in run_tests
    old_config = self.setup_databases(aliases=databases)
  File "/home/runner/work/rest_api/rest_api/env/lib/python3.9/site-packages/django/test/runner.py", line 604, in setup_databases
    return _setup_databases(
  File "/home/runner/work/rest_api/rest_api/env/lib/python3.9/site-packages/django/test/utils.py", line 177, in setup_databases
    connection.creation.clone_test_db(
  File "/home/runner/work/rest_api/rest_api/env/lib/python3.9/site-packages/django/db/backends/base/creation.py", line 216, in clone_test_db
    self._clone_test_db(suffix, verbosity, keepdb)
  File "/home/runner/work/rest_api/rest_api/env/lib/python3.9/site-packages/djongo/creation.py", line 30, in _clone_test_db
    shutil.rmtree('dump')
  File "/opt/hostedtoolcache/Python/3.9.4/x64/lib/python3.9/shutil.py", line 709, in rmtree
    onerror(os.lstat, path, sys.exc_info())
  File "/opt/hostedtoolcache/Python/3.9.4/x64/lib/python3.9/shutil.py", line 707, in rmtree
    orig_st = os.lstat(path)
FileNotFoundError: [Errno 2] No such file or directory: 'dump'
make: *** [Makefile:35: test] Error 1
Error: Process completed with exit code 2.
```

**Comment by hugofpaiva** [comment]
While trying to make the tests run on a Docker Container on my local machine to use this container on GitHub Actions I came across the same error...

After some try and error I found that the ''--parallel'' option was causing the issue while running ''python3 manage.py test --verbosity=0 --parallel --failfast''. Maybe ''Djongo'' has some problem running tests in parallel, but it is still strange that I could run it in parallel in my local machine without using a Docker Container 🤔

Anyway, my problem is fixed, but I will still keep the issue open since ''Djongo'' has some problem with parallel tests.

**Comment by jie8357IOII** [comment]
This is because it didn't [connect to mongodb test database in parallel mode](https://github.com/nesdis/djongo/blob/master/djongo/creation.py#L15-L21).

After I remove the `--quiet` parameter, I got the following error
`root@f1d04b6fdb1b:/code# mongodump --db test_athena_labeling
2021-08-05T11:15:10.311+0700    Failed: error connecting to db server: no reachable servers`

**Comment by florealcab** [comment]
Same problem for me. Could be cool if this could be fixed...

**Comment by florealcab** [comment]
I made a pull request to fix this issue. The problem is that the code suppose on the database cloning that mongo is on localhost, which is not the case if we use an external docker container for example. So I add the host info on the that part of code.

Waiting for the merge of this request, you can use the fork I published on pypi: djongo-bn (version 1.3.6.1 match exactly the code at the date of 18.03.2022 + only the fix)
